pkgs/development/python-modules/causal-conv1d/default.nix

```diff
@@ -0,0 +1,61 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  ninja,
+  torch,
+  cudaPackages,
+  rocmPackages,
+  config,
+  cudaSupport ? config.cudaSupport,
+  which,
+}:
+
+buildPythonPackage rec {
+  pname = "causal-conv1d";
+  version = "1.4.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "Dao-AILab";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-p5x5u3zEmEMN3mWd88o3jmcpKUnovTvn7I9jIOj/ie0=";
+  };
+
+  build-system = [
+    ninja
+    which
+  ];
+
+  buildInputs = (
+    lib.optionals cudaSupport (
+      with cudaPackages;
+      [
+        cuda_cudart # cuda_runtime.h, -lcudart
+        cuda_cccl
+        libcusparse # cusparse.h
+        libcusolver # cusolverDn.h
+        cuda_nvcc
+        libcublas
+      ]
+    )
+  );
+
+  dependencies = [
+    torch
+  ];
+
+  pythonImportsCheck = [ "causal_conv1d" ];
+
```

Comment:
Could you enable tests with `pytestCheckHook`?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pytestcheckhook-using-pytestcheckhook

Reply:
I've checked, neither introduced packages actually supports cpu only execution, so I've marked the package as broken without CUDA. I'm going to leave tests disabled because I don't have a nixos machine with cuda to experiment on.

Reply:
That seems fair enough; shall we leave a comment explaining the situation?

Reply:
I've added a comment indicating the tests are disabled.